### ISSUE TITLE
Use the 'await' expression instead of callbacks in Redstone

### ISSUE THREAD

**Issue by Emasoft** [redstone-dart/redstone#37]
The initial support for the 'await' expression in Dart has been commited, as you can read here:
https://code.google.com/p/dart/source/detail?r=39345
Using 'await' is a tremendous advantage for server side application development. Asynchronous programming becomes very easy to implement.
Can you consider to refactor Redstone to making use of it in a new branch, even if it will be released officially in Redstone only in a later version when the Dart team will complete it? I'm starting a new project using Redstone and I'd like to write from the beginning my code to make use of the 'async...await' pattern, to avoid having to rewrite most of my code. Thank you for this wonderful framework anyway!

**Comment by luizmineo** [comment]
v0.6 (which I hope to publish on github soon) uses the new async/await and enum features, which makes the code easier to understand and maintain. Although, note that it doesn't stop you from using these features in your application right now, with the current (v0.5) stable version.

**Comment by luizmineo** [comment]
The first alpha release of Redstone v0.6 is available for test: https://github.com/luizmineo/redstone.dart/tree/v0.6

**Comment by graphicbeacon** [comment]
Hi, in response to your previous comment, I am having issues when I use async await in 0.5.21+. The code breaks whenever a completeError is returned from a Completer instance.

```
Failed to execute simple_app.SimpleAppApi.login - Wrong username or password.
```

"Wrong username and password" is the error message returned from the completer. On the client I get a 500 error message.

Code looks like this:

```
// File with login functionality
Future login(Map<String, String> credentials) {
        var completer = new Completer();
        retrieveDataFromDatabase().then(() {
            // Create session token
            var sessionToken = 'blahblah';

            completer.complete(sessionToken);
        })
                .catchError(() {
            completer.completeError("Wrong username or password.");
        });

        return completer.future;
    }
```

And the second file:

```
@app.Route('/login', methods: const [app.POST])
    login(@app.Body(app.JSON) Map payload) async {

        var credentials = payload["data"];
        var loginResponse = await new AuthenticationService(user).login(credentials);
        ......
```

As soon as the await at var loginResponse completes with an error(completer.completeError) this exception is thrown:

```
SEVERE: 2015-05-25 20:21:58.510: Failed to execute simple_app.SimpleAppApi.login - Wrong username or password.
dart:isolate  _RawReceivePortImpl._handleMessage
```

As a result I'm not able to capture this error and craft a response detailing this. Problem does not happen when I replace completeError with complete, but I'm not too keen on that.

I understand if you are not releasing any fixes for 0.5.21+ to focus on 0.6, I'll just use the workaround. At least this is something to test for in 0.6.

**Comment by Pacane** [comment]
0.6 was released.
